ocaml-lsp-server/src/diagnostics.ml

```diff
@@ -0,0 +1,64 @@
+open Import
+
+module Uri = struct
+  include Uri
+
+  let compare x y = Ordering.of_int (Uri.compare x y)
+end
+
+module Uri_c = Comparable.Make (Uri)
+module Uri_set = Uri_c.Set
+
+type t =
+  { dune : (int, Uri.t * Diagnostic.t) Table.t
+  ; merlin : (Uri.t, Diagnostic.t list) Table.t
+  ; send : PublishDiagnosticsParams.t -> unit Fiber.t
+  ; mutable dirty_uris : Uri_set.t
```

Comment:
Not sure if that will be accessed concurrently, but we should probably lock the access to `dirty_uris` in `Diagnostics`, to be safe.

Reply:
I don't think using mutexes defensively is a good idea. They can introduce their own problems like additional deadlocks. After all, we are using fiber to (mostly) avoid the issues of multi threaded programming. We can use `Fiber.Mutex.t` when we want to do something atomically across more than 1 fiber, but it's clear we don't need it here.